File: Programmers/Level1/최소직사각형/최소직사각형.js
function solution(sizes) {
    const width = [];
    const height = [];
    sizes.map((e) => {
        e.sort((a, b) => a - b);
        width.push(e[1]);
        height.push(e[0]);
    });
    console.log(Math.max(...width) * Math.max(...height));
}

// 다른 풀이
function solution(sizes) {
    return (
        Math.max(...sizes.map((e) => Math.max(...e))) *
        Math.max(...sizes.map((e) => Math.min(...e)))
    );
}

File: Programmers/Level1/최소직사각형/최소직사각형.py
def solution(sizes):
    width = []
    height = []
    for s in sizes:
        if s[0] < s[1]:
            s[0], s[1] = s[1], s[0]
        width.append(s[0])
        height.append(s[1])

    return max(width) * max(height)

# 다른 풀이
def solution(sizes):
    return max(max(x) for x in sizes) * max(min(x) for x in sizes)
